server/src/main/java/org/folio/reservoir/server/OaiPmhClientService.java

```diff
@@ -520,22 +515,30 @@ Future<Boolean> ingestRecord(
     }
   }
 
-  Future<HttpClientResponse> listRecordsRequest(JsonObject config) {
-    RequestOptions requestOptions = new RequestOptions();
-    requestOptions.setMethod(HttpMethod.GET);
-    requestOptions.setHeaders(getHttpHeaders(config));
-    QueryStringEncoder enc = new QueryStringEncoder(config.getString("url"));
-    enc.addParam("verb", "ListRecords");
-    if (!addQueryParameterFromConfig(enc, config, RESUMPTION_TOKEN_LITERAL)) {
-      addQueryParameterFromConfig(enc, config, "from");
-      addQueryParameterFromConfig(enc, config, "until");
-      addQueryParameterFromConfig(enc, config, "set");
-      addQueryParameterFromConfig(enc, config, "metadataPrefix");
+  Future<OaiResponse<JsonObject>> listRecordsRequest(OaiPmhStatus job) {
+    JsonObject config = job.getConfig();
+    XmlMetadataStreamParser<JsonObject> metadataParser = new XmlMetadataParserMarcInJson();
+    OaiRequest<JsonObject> oaiRequest = new OaiHttpRequest<>(httpClient, config.getString("url"),
+        metadataParser, config.getBoolean("xmlFixing", false), getHttpHeaders(config));
+    String resumptionToken = config.getString(RESUMPTION_TOKEN_LITERAL);
+    if (resumptionToken != null) {
+      oaiRequest.token(resumptionToken);
+    } else {
+      oaiRequest
+          .from(config.getString("from"))
+          .until(config.getString("until"))
+          .set(config.getString("set"))
+          .metadataPrefix(config.getString("metadataPrefix"));
     }
-    addQueryParameterFromParams(enc, config.getJsonObject("params"));
-    String absoluteUri = enc.toString();
-    requestOptions.setAbsoluteURI(absoluteUri);
-    return httpClient.request(requestOptions).compose(HttpClientRequest::send);
+    JsonObject params = config.getJsonObject("params");
```

Comment:
Move to OaiHttpReqest "fromConfig" or "fromJson" method.

Reply:
I'm not sure we want this at all as this is non-standard OAI. The limit however is important for Solr stuff.

Reply:
Are you saying that you want to turn OaiPmhClient service into a general service for managing not only OAI-PMH clients but also other things like SOLR export clients (aka outlets)? I see the similarity that all of these things are "jobs" but I wouldn't want all of them to be available under the same endpoint.